lib/auth/windows/windows.go

```diff
@@ -105,17 +120,38 @@ type AuthInterface interface {
 	GetClusterName(opts ...services.MarshalOption) (types.ClusterName, error)
 }
 
-// GenerateCredentials generates a private key / certificate pair for the given
+// GenerateCredentialsRequest are the request parameters for
+// generating a windows cert/key pair
+type GenerateCredentialsRequest struct {
+	// Username is the Windows username
+	Username string
+	// Domain is the Windows domain
+	Domain string
+	// TTL is the ttl for the certificate
+	TTL time.Duration
+	// ClusterName is the local cluster name
+	ClusterName string
+	// ActiveDirectorySID is the SID of the Windows user
+	// specified by Username. If specified (non-nil), it is
+	// encoded in the certificate per https://go.microsoft.com/fwlink/?linkid=2189925.
+	ActiveDirectorySID *string
```

Comment:
How about plain old `string` where empty is unspecified?

Reply:
9e898b30c4fccc2a25f74cfc06eb4e1e481bc57d